interop-testing/src/main/java/io/grpc/testing/integration/StressTestClient.java

```diff
@@ -0,0 +1,574 @@
+              + "\n    List of <testcase,weight> tuples. Weight is the relative frequency at which"
+              + " testcase is run."
+              + "\n    Valid Testcases:"
+              + validTestCasesHelpText()
+              + "\n  --test_duration-secs=SECONDS   '-1' for no limit. Default: " + c.durationSecs
+              + "\n  --num_channels_per_server=INT  Number of connections to each server address."
+              + " Default: " + c.channelsPerServer
+              + "\n  --num_stubs_per_channel=INT    Default: " + c.stubsPerChannel
+              + "\n  --metrics_port=PORT            Listening port of the metrics server."
+              + " Default: " + c.metricsPort
+      );
+      System.exit(1);
+    }
+  }
+
+  @VisibleForTesting
+  void startMetricsService() throws IOException {
+    Preconditions.checkState(!shutdown, "client was shutdown.");
+
+    metricsServer = ServerBuilder.forPort(metricsPort)
+        .addService(MetricsServiceGrpc.bindService(new MetricsServiceImpl()))
+        .build()
+        .start();
+  }
+
+  @VisibleForTesting
+  void runStressTest() throws ExecutionException, InterruptedException {
+    Preconditions.checkState(!shutdown, "client was shutdown.");
+
+    int numChannels = addresses.size() * channelsPerServer;
+    int numThreads =  numChannels * stubsPerChannel;
+    threadpool = MoreExecutors.listeningDecorator(newFixedThreadPool(numThreads));
+    try {
+      int server_idx = -1;
+      for (InetSocketAddress address : addresses) {
+        server_idx++;
+        for (int i = 0; i < channelsPerServer; i++) {
+          ManagedChannel channel = createChannel(address);
+          channels.add(channel);
+          for (int j = 0; j < stubsPerChannel; j++) {
+            String gaugeName =
+                String.format("/stress_test/server_%d/channel_%d/stub_%d/qps", server_idx, i, j);
+            Worker worker =
+                new Worker(channel, testCaseWeightPairs, durationSecs, gaugeName);
+
+            workerFutures.add(threadpool.submit(worker));
+          }
+        }
+      }
+    } catch (Exception e) {
+      log.log(Level.WARNING, "The stress test client encountered an error!", e);
+    }
+  }
+
+  @VisibleForTesting
+  void blockUntilStressTestComplete() throws Exception {
+    Preconditions.checkState(!shutdown, "client was shutdown.");
+
+    ListenableFuture<?> f = Futures.allAsList(workerFutures);
+    if (durationSecs < 0) {
+      // -1 indicates that the stress test runs "forever".
+      f.get();
+    } else {
+      f.get(durationSecs + WORKER_GRACE_PERIOD_SECS, TimeUnit.SECONDS);
+    }
+  }
+
+  @VisibleForTesting
+  void shutdown() {
+    if (shutdown) {
+      return;
+    }
+    shutdown = true;
+
+    for (ManagedChannel ch : channels) {
+      try {
+        ch.shutdownNow();
+      } catch (Throwable t) {
+        log.log(Level.WARNING, "Error shutting down channel!", t);
+      }
+    }
+
```

Comment:
You should probably await termination here, since shutdownNow is not implemented as advertised.